wagtail/models/i18n.py

```diff
@@ -205,22 +204,17 @@ def has_translation(self, locale):
             self.get_translations(inclusive=True).filter(locale_id=pk(locale)).exists()
         )
 
-    def copy_for_translation(self, locale):
+    def copy_for_translation(self, locale, exclude_fields=None):
         """
         Creates a copy of this instance with the specified locale.
 
         Note that the copy is initially unsaved.
         """
-        translated, child_object_map = _copy(self)
-        translated.locale = locale
-
-        # Update locale on any translatable child objects as well
-        # Note: If this is not a subclass of ClusterableModel, child_object_map will always be '{}'
-        for (_child_relation, _old_pk), child_object in child_object_map.items():
-            if isinstance(child_object, TranslatableMixin):
-                child_object.locale = locale
-
-        return translated
```

Comment:
If this change is too drastic for 4.2 RC, we can also just copy the following parts and not introduce the `CopyForTranslationAction` class:

```py
        # Make sure the copy includes the latest changes, including draft
        if isinstance(object, DraftStateMixin):
            object = object.get_latest_revision_as_object()

        exclude_fields = (
            getattr(object, "default_exclude_fields_in_copy", [])
            + getattr(object, "exclude_fields_in_copy", [])
            + (exclude_fields or [])
        )
        translated, child_object_map = _copy(object, exclude_fields=exclude_fields)
```